src/main.rs

```diff
@@ -79,31 +79,56 @@ fn main() {
         FileConfig::default()
     };
 
-    let cfgfile =
-        args.configfile.as_ref().map_or_else(|| Path::new("Config.toml"), |p| p.as_path());
-    let file = if cfgfile.is_file() {
-        let content = fs::read_to_string(cfgfile).expect("error reading existing config file");
-        toml::from_str(&content).expect("invalid config")
-    } else {
-        FileConfig::default()
+    // try to find a heradoc.toml
+    // 1. passed via commandline
+    // 2. next to the input file
+    // 3. somewhere in the parent folders of the input file
+    let cfgfile_folder = match args.configfile.as_ref() {
+        Some(file) => Some(file.parent().expect("configfile passed as argument should point to a file and not the root-folder").to_owned()),
```

Comment:
I see we have enough of these panic-unwraps above as well. Oh well, it might be time for a proper error with variants.